src/routes.c

```diff
@@ -437,6 +447,92 @@ static wolfsentry_errcode_t wolfsentry_route_init(
     WOLFSENTRY_RETURN_OK;
 }
 
+static wolfsentry_errcode_t wolfsentry_route_init_by_exports(
+    struct wolfsentry_event *parent_event,
+    const struct wolfsentry_route_exports *route_exports,
+    size_t data_addr_offset,
+    size_t data_addr_size,
+    struct wolfsentry_route *new)
+{
+    if (route_exports->private_data_size > data_addr_offset)
+        WOLFSENTRY_ERROR_RETURN(INVALID_ARG);
+    if (data_addr_size < WOLFSENTRY_BITS_TO_BYTES(route_exports->remote.addr_len) + WOLFSENTRY_BITS_TO_BYTES(route_exports->local.addr_len))
+        WOLFSENTRY_ERROR_RETURN(BUFFER_TOO_SMALL);
+    if (data_addr_size > MAX_UINT_OF(uint16_t))
+        WOLFSENTRY_ERROR_RETURN(NUMERIC_ARG_TOO_BIG);
+    if ((unsigned)data_addr_offset > MAX_UINT_OF(uint16_t))
+        WOLFSENTRY_ERROR_RETURN(NUMERIC_ARG_TOO_BIG);
+    if (! (route_exports->flags & (WOLFSENTRY_ROUTE_FLAG_DIRECTION_IN | WOLFSENTRY_ROUTE_FLAG_DIRECTION_OUT)))
+        WOLFSENTRY_ERROR_RETURN(INVALID_ARG);
+
+    /* make sure wildcards are sensical. */
+    if (((route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_FAMILY_WILDCARD) &&
+         ((! (route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_REMOTE_ADDR_WILDCARD)) ||
+          (! (route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_LOCAL_ADDR_WILDCARD)) ||
+          (! (route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_PROTO_WILDCARD)))) ||
+        ((route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_PROTO_WILDCARD) &&
+         ((! (route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_REMOTE_PORT_WILDCARD)) ||
+          (! (route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_LOCAL_PORT_WILDCARD)))))
+        WOLFSENTRY_ERROR_RETURN(INVALID_ARG);
+
+    memset(new,0,offsetof(struct wolfsentry_route, data));
```

Comment:
space after commas for readability

Reply:
fixed, and I found and fixed another one just like it in `wolfsentry_route_init()`.